### PR TITLE
fix: add empty `value` check in getRequestBodyExamples()

### PR DESCRIPTION
**Pull request by darrenyong**
| 🚥 Resolves ISSUE_ID |
| :------------------- |

## 🧰 Changes
There is a weird case in our main repo where we initialize `Webhooks` with an empty `requestBodyExamples` with the following structure:
```
requestBodyExamples: [
  {
    mediaType: 'application/json',
    examples: [{
      value: undefined
    }],
  },
];
```

This causes `Operation.getRequestBodyExamples()` to return with an empty request example even though we should be creating an example for them. This adds an extra check so that we avoid returning the empty example and proceed to the subsequent `getRequestBodyExamples()` call that generates an example for the user.

## 🧬 QA & Testing

Provide as much information as you can on how to test what you've done.

### CODE REVIEW

File: packages/oas/src/operation/index.ts [hunk 1]
```diff
@@ -708,7 +708,9 @@ export class Operation {
    *
    */
   getRequestBodyExamples(): RequestBodyExamples {
-    if (this.requestBodyExamples) {
+    const nonEmptyRequestExample = this.requestBodyExamples?.[0]?.examples?.[0].value;
```
Comment (darrenyong) [inline packages/oas/src/operation/index.ts:711]: this is ugly :(

File: packages/oas/test/operation/lib/get-requestbody-examples.test.ts [hunk 2]
```diff
@@ -5,20 +5,53 @@ import cleanStringify from '../../__fixtures__/json-stringify-clean.js';
...
+  webhooksOas = await import('@readme/oas-examples/3.1/json/webhooks.json').then(r => r.default).then(Oas.init);
+  await webhooksOas.dereference();
 });
 
 test('should return early if there is no request body', () => {
   const operation = operationExamples.operation('/nothing', 'get');
   expect(operation.getRequestBodyExamples()).toStrictEqual([]);
 });
 
+test('webhooks', () => {
```
Comment (erunion) [inline packages/oas/test/operation/lib/get-requestbody-examples.test.ts:26]: can you give this a description?

File: packages/oas/test/operation/lib/get-requestbody-examples.test.ts [hunk 3]
```diff
@@ -5,20 +5,53 @@ import cleanStringify from '../../__fixtures__/json-stringify-clean.js';
...
+  const webhookOperation = webhooksOas.operation('newPet', 'post', { isWebhook: true });
+  webhookOperation.requestBodyExamples = [
+    {
+      mediaType: 'application/json',
+      examples: [
+        {
+          value: undefined,
```
Comment (erunion) [inline packages/oas/test/operation/lib/get-requestbody-examples.test.ts:33]: are you able to recreate this without having to set `.requestBodyExamples` to the operation instance?

**Review by erunion — DISMISSED** [review]

File: packages/oas/src/operation/index.ts (hunk 1 above)
Comment (kanadgupta) [reply packages/oas/src/operation/index.ts:711]: lol nauuur it's great! would this also solve the problem at hand? roughly the same from an ugliness standpoint but i think it's safer since it would allow falsy values like `0`, `false`, etc. (might also be good to have a test case for falsy values too!)

```suggestion
    const isRequestExampleValueDefined = typeof this.requestBodyExamples?.[0]?.examples?.[0].value !== 'undefined';
```

File: packages/oas/test/operation/lib/get-requestbody-examples.test.ts (hunk 3 above)
Comment (kanadgupta) [reply packages/oas/test/operation/lib/get-requestbody-examples.test.ts:33]: i'd be cool with creating a openapi definition fixture in this repo for this! don't think it's noteworthy enough to publish a new version of `oas-examples` or anything like that

File: packages/oas/test/operation/lib/get-requestbody-examples.test.ts (hunk 2 above)
Comment (kanadgupta) [inline packages/oas/test/operation/lib/get-requestbody-examples.test.ts:26]: something like this perhaps?

```suggestion
test('should generate a request example for a webhook operation with an undefined example value', () => {
```

**Review by kanadgupta — CHANGES_REQUESTED** [review]
whoa incredible find here 🫨 small suggestion to account for a potential edge case but overall happy with this!

File: packages/oas/test/operation/lib/get-requestbody-examples.test.ts [hunk 4]
```diff
@@ -5,20 +5,54 @@ import cleanStringify from '../../__fixtures__/json-stringify-clean.js';
...
+
+  webhooksOas = await import('@readme/oas-examples/3.1/json/webhooks.json').then(r => r.default).then(Oas.init);
 });
 
 test('should return early if there is no request body', () => {
   const operation = operationExamples.operation('/nothing', 'get');
   expect(operation.getRequestBodyExamples()).toStrictEqual([]);
 });
 
+test('should generate a request example for an operation with an undefined example value', async () => {
```
Comment (kanadgupta) [inline packages/oas/test/operation/lib/get-requestbody-examples.test.ts:25]: now that we understand the underlying issue a bit better, i think this would be a better title:

```suggestion
test('should re-intialize the request examples after the oas is dereferenced', async () => {
```

**Review by kanadgupta — APPROVED** [review]
small test renaming suggestion, otherwise LGTM!

**Review by kanadgupta — APPROVED** [review]
